private/buf/bufformat/formatter.go

```diff
@@ -2023,11 +2016,99 @@ func (f *formatter) writeTrailingEndComments(comments ast.Comments) {
 		if i > 0 || comment.LeadingWhitespace() != "" {
 			f.Space()
 		}
-		f.WriteString(strings.TrimSpace(comment.RawText()))
+		f.writeComment(comment.RawText())
 	}
 	f.P()
 }
 
+func (f *formatter) writeComment(comment string) {
+	if strings.HasPrefix(comment, "/*") && newlineCount(comment) > 0 {
+		lines := strings.Split(comment, "\n")
+		// find minimum indent, so we can make all other lines relative to that
+		minIndent := -1 // sentinel that means unset
+		// start at 1 because line at index zero starts with "/*", not whitespace
+		for i := 1; i < len(lines); i++ {
+			indent, ok := computeIndent(lines[i])
+			if ok && (minIndent == -1 || indent < minIndent) {
+				minIndent = indent
+			}
+		}
+		if minIndent < 0 {
+			// This shouldn't be necessary.
+			// But we do it just in case, to avoid possible panic
+			minIndent = 0
+		}
+		for i, line := range lines {
+			trimmedLine := strings.TrimSpace(line)
+			if trimmedLine == "" || trimmedLine == "*/" {
+				line = trimmedLine
+			} else {
+				// we only trim space from the right; for the left,
+				// we unindent based on indentation found above.
+				line = unindent(line, minIndent)
+				line = strings.TrimRightFunc(line, unicode.IsSpace)
+			}
+			if i > 0 && line != "*/" {
+				line = "   " + line
+			}
+			if i != len(lines)-1 {
+				f.P(line)
+			} else {
+				// for last line, we don't use P because we don't
+				// want to print a trailing newline
+				f.Indent(nil)
+				f.WriteString(line)
+			}
+		}
+	} else {
+		f.Indent(nil)
+		f.WriteString(strings.TrimSpace(comment))
+	}
+}
+
+func unindent(s string, unindent int) string {
+	for i, r := range s {
+		if unindent == 0 {
+			return s[i:]
+		}
+		if unindent < 0 {
+			// removing tab-stop unindented too far, so we
+			// add back some spaces to compensate
+			return strings.Repeat(" ", -unindent) + s[i:]
+		}
+
+		switch r {
+		case ' ':
+			unindent--
+		case '\t':
+			unindent -= 4
+		default:
+			return s[i:]
+		}
+	}
+	// nothing but whitespace...
+	return ""
+}
+
+func computeIndent(s string) (int, bool) {
+	if strings.TrimSpace(s) == "*/" {
+		return 0, false
+	}
+	indent := 0
+	for _, r := range s {
+		switch r {
+		case ' ':
+			indent++
+		case '\t':
+			indent += 4
```

Comment:
Is this normal for proto? Not 8? It's at least some magic given common editors will prefix fill with tabs and then add spaces, but how many tabs and spaces to use depends on the configured tab width.

Reply:
Just like you said: this depends on the editor (nothing to do with proto). I can change it to 8. Most editors I've used (like sublime, IntelliJ, and VSCode I think, too) use 4. But editors like vim or emacs will use tab stops with width 8. I guess it's probably less surprising to just use the standard 8, even if someone's editor is configured otherwise, so I'll change it.

Reply:
Changed to eight. I also realized that incrementing by 8 is not correct since tab stops are at fixed intervals of 8, and not simply 8 sequential spaces. So I fixed that, too.